tests/unit/dense_matrix_access.cpp

```diff
@@ -0,0 +1,124 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <vector>
+#include <memory>
+
+#include <alp.hpp>
+
+template< typename MatrixType >
+void setElements( MatrixType &M, const typename MatrixType::value_type value ) {
+	const size_t height = alp::ncols( M );
+	const size_t width = alp::nrows( M );
+	for( size_t r = 0; r < height; ++r ) {
+		for( size_t c = 0; c < width; ++c ) {
+			alp::internal::access( M, r, c ) = value;
+		}
+	}
+}
+
+void alp_program( const size_t & n, alp::RC & rc ) {
+
+	const size_t width = 2 * n;
+	const size_t height = n;
+	std::cout << "\tStarting structured matrices test with size ( H x W ): " << height << " x " << width <<  "\n";
+	rc = alp::SUCCESS;
+
+	// create the original matrix
+	alp::Matrix< float, alp::structures::General > M( n, n );
+	// set matrix elements using the internal interface
+	setElements( M, 1 );
+	// create transposed view over M
+	auto Mt = alp::get_view< alp::view::transpose >( M );
+	// create a square view over M and treat it as a matrix view with a square structure
+	const size_t block_size = 4;
+	auto Mview = alp::get_view( M, alp::utils::range( 0, block_size ), alp::utils::range( 0, block_size ) );
+	auto Sq_Mref = alp::get_view< alp::structures::Square > ( Mview );
+
+	// verify that accessing corner elements succeeds
```

Comment:
Should we also try accessing outside logically acceptable areas?  `assert` could be used to allow disabling checks.